### ISSUE TITLE
Exception when deserialising enums with newer Python versions

### ISSUE THREAD

**Issue by haraldF** [ndsev/zserio#619]
**Zserio version and language**
Zserio: 2.14.0
Language: Python

**Describe the bug**
It seems there's an issue with deserialising enums with Python 3.12.3. Downgrading to Python 3.9.6 makes the code work (see snippet below)

**How to reproduce**

```python
import zserio
import registry.api as registry_api
import requests

registry_url = "https://api.nds.live/registry"
api_key = "XXX"

response = requests.get(f"{registry_url}/getAllNodesAllSystems?api-key={api_key}")
reader = zserio.BitStreamReader(response.content)
registry = registry_api.node.NodeList.from_reader(reader)
```

Exception with Python 3.12: `(1, None) is not a valid NodeType`

**Expected behavior**
No exception in Python 3.12.

**Comment by mikir** [comment]
Thank you very much for a report!

You are right, we were hit by incompatible change in Python enumerations done in the patch version 3.12.3. Python version 3.12.2 and older versions work without any problem.

We have already fixed it within the scope of the issue #595. Unfortunatelly, the fix is not in the latest Zserio version 2.14. In the meantime, you can download fixed binaries from the [GitHub Actions](https://github.com/ndsev/zserio/actions/runs/9190063576/artifacts/1527262589) or you can downgrade your Python to the version 3.12.2.

**Comment by mikir** [comment]
This is a duplicate to #595.
